src/Digdir.Domain.Dialogporten.Infrastructure/Altinn/OrganizationRegistry/OrganizationRegistryClient.cs

```diff
@@ -0,0 +1,66 @@
+using System.Diagnostics;
+using System.Net.Http.Json;
+using Digdir.Domain.Dialogporten.Application.Externals;
+using Digdir.Domain.Dialogporten.Infrastructure.Common.Extensions;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace Digdir.Domain.Dialogporten.Infrastructure.Altinn.OrganizationRegistry;
+
+internal class OrganizationRegistryClient : IOrganizationRegistry
+{
+    private const string OrgShortNameReferenceCacheKey = "OrgShortNameReference";
+    private static readonly DistributedCacheEntryOptions _oneDayCacheDuration = new() { AbsoluteExpiration = DateTimeOffset.UtcNow.AddDays(1) };
+    private static readonly DistributedCacheEntryOptions _zeroCacheDuration = new() { AbsoluteExpiration = DateTimeOffset.MinValue };
+
+    private readonly IDistributedCache _cache;
+    private readonly HttpClient _client;
+
+    public OrganizationRegistryClient(HttpClient client, IDistributedCache cache)
+    {
+        _client = client ?? throw new ArgumentNullException(nameof(client));
+        _cache = cache ?? throw new ArgumentNullException(nameof(cache));
+    }
+    public async Task<string?> GetOrgShortName(string orgNumber, CancellationToken cancellationToken)
+    {
+        var orgShortNameByOrgNumber = await _cache.GetOrAddAsync(
```

Comment:
Ikke en del av denne PR-en, men GetOrAddAsync har race conditions knyttet til cache misses, og burde kanskje ha noen form for locking?

Reply:
Enig i det, ser det kan bli krøll her. Kanskje best å bryte ut i separat issue?